### PR TITLE
Feature flag refactoring (part 2)

### PR DESCRIPTION
**Pull request by pcapriotti**
Follow-up to #4181.

This PR reorganises and simplifies the servant endpoints related to features.

https://wearezeta.atlassian.net/browse/WPB-10323

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/wire-api/src/Wire/API/Routes/Internal/Galley.hs [hunk 1]
```diff
@@ -56,128 +56,40 @@ import Wire.API.Team.Member
 import Wire.API.Team.SearchVisibility
 import Wire.API.User.Client
 
-type LegalHoldFeatureStatusChangeErrors =
```
Comment (elland) [inline libs/wire-api/src/Wire/API/Routes/Internal/Galley.hs:59]: 👏

File: libs/wire-api/src/Wire/API/Routes/Named.hs [hunk 2]
```diff
@@ -147,3 +147,7 @@ infixr 4 ::>
 type instance
   x ::> (Named name api) =
     Named name (x :> api)
+
+type instance
```
Comment (elland) [inline libs/wire-api/src/Wire/API/Routes/Named.hs:151]: Could we add some documentation to `::>`?

**Review by elland — APPROVED** [review]
Small suggestions.

File: libs/wire-api/src/Wire/API/Routes/Internal/Galley.hs (hunk 1 above)
Comment (MangoIV) [reply libs/wire-api/src/Wire/API/Routes/Internal/Galley.hs:59]: I did this fix at some point already, but it got discarded.

**Review by MangoIV — APPROVED** [review]
no objections

**Pull request merged**